cmd/network/verification.go

```diff
@@ -214,6 +219,10 @@ func (e *EgressVerification) setup(ctx context.Context) (*aws.Config, error) {
 	}
 	e.log = logger
 
+	if validArches := []string{"", "x86", "x86_64", "amd64", "arm", "arm64", "aarch64"}; !slices.Contains(validArches, e.CpuArchName) {
+		return nil, fmt.Errorf("%s is not a valid architecture - must use one of %s", e.CpuArchName, strings.Join(validArches, ","))
+	}
+
```

Comment:
This check isn't really necessary as ` cpu.ArchitectureByName(e.CpuArchName)` will return an empty `CPUArchitecture{}` that will fail `IsValid()` and throw "unknown CPU architecture" (lines 319-322). I'd suggest removing this so that you don't have to import `slices` and so that there's only one source of truth when it comes to the arch's that the verifier supports

Reply:
I want us to error out early on here. Otherwise we're going through backplane to grab the cluster credentials, setting up an AWS config, fetching info, etc. This wastes a lot of resources if we know right from the get go that we have an invalid flag value.

Reply:
I would want network-verifier to expose an arch validation function (if it's too costly), it doesn't seem to make sense for osdctl to pre-validate this before passing it to network-verifier.

Reply:
Idea incoming in a new commit.

Reply:
Can do low-cost early arch evaluation by calling cpu.ArchitectureByName(e.CpuArchName).IsValid() early on

Reply:
@abyrne55 +1, this is what I ended up going with.